### PR TITLE
Puzzle4

### PR DESCRIPTION
**Pull request by iamsammak**
I'm going to delete/closeout PR #58 since this PR replaces that one. Double check my work. But main updates are using getCurrentStage()

### CODE REVIEW

File: frontend/components/puzzle4.jsx [hunk 1]
```diff
@@ -34,16 +36,27 @@ class Puzzle4 extends React.Component {
     });
   }
 
+  submitAnswer() {
+    const answer = this.state.value;
+    const gameState = this.props.gameState;
+    let stage = getCurrentStage(gameState);
```
Comment (ihurrahi) [inline frontend/components/puzzle4.jsx]: this can be const

File: frontend/components/puzzle4.jsx [hunk 2]
```diff
@@ -2,8 +2,11 @@ import React from 'react';
 
 import PuzzleHeader from './puzzleHeader';
 
-import {
-  KeypadContainer, Button } from './buttonContants';
+import { KeypadContainer, Button,
+  SubmitButton, ClearButton } from './buttonContants';
+import getCurrentStage from '../helpers';
```
Comment (ihurrahi) [inline frontend/components/puzzle4.jsx]: does this need to be a named import?

File: frontend/components/puzzle4.jsx (hunk 1 above)
Comment (iamsammak) [reply frontend/components/puzzle4.jsx]: fixed

File: frontend/components/puzzle4.jsx (hunk 2 above)
Comment (iamsammak) [reply frontend/components/puzzle4.jsx]: updated

**Pull request merged**